Text inputs on logic-masters.de overflow their containers

pxlcoder opened the pull request:
#### 3e56a5447cd445cd703ce932e003e7acba710859
<pre>
Text inputs on logic-masters.de overflow their containers
<a href="https://bugs.webkit.org/show_bug.cgi?id=298966">https://bugs.webkit.org/show_bug.cgi?id=298966</a>
<a href="https://rdar.apple.com/159975950">rdar://159975950</a>

Reviewed by Abrar Rahman Protyasha.

logic-masters.de uses the size attribute to set the width of text inputs and
then attempts to restrict the input width to the container using either
`width: 100%` or `max-width: 100%`.

However, this results in the input still being slightly larger than expected
due to the default padding of native inputs. The overflow is even worse on
macOS and iOS 26, since the default padding of inputs was increased.

The site should be using `box-sizing: border-box;` here to ensure that padding
is accounted for in the max-width. Until they do, quirk the behavior to fix the
issue.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsTextInputBoxSizingBorderBoxQuirk const):
(WebCore::handleLogicMastersQuirks):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Canonical link: <a href="https://commits.webkit.org/300048@main">https://commits.webkit.org/300048@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2270a39c381b312dd4ed8faacd2b54cf759ccad8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121147 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40843 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31501 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127570 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73228 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8bef1e3e-8873-4ef7-a16d-a2641d3efbf3) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123023 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41545 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49422 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92032 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/61234 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f849bace-0942-4d2b-8621-3a8117839a90) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124099 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33174 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108594 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72708 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c0517eb1-3cb5-4319-b9af-4140acf77cdd) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32199 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26697 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71159 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102678 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26876 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130419 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48074 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36539 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100630 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48442 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104761 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100534 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45936 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23994 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44768 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19220 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47932 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53645 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47403 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50750 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49087 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2270a39c381b312dd4ed8faacd2b54cf759ccad8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50819

pxlcoder committed:
Text inputs on logic-masters.de overflow their containers
https://bugs.webkit.org/show_bug.cgi?id=298966
rdar://159975950

Reviewed by Abrar Rahman Protyasha.

logic-masters.de uses the size attribute to set the width of text inputs and
then attempts to restrict the input width to the container using either
`width: 100%` or `max-width: 100%`.

However, this results in the input still being slightly larger than expected
due to the default padding of native inputs. The overflow is even worse on
macOS and iOS 26, since the default padding of inputs was increased.

The site should be using `box-sizing: border-box;` here to ensure that padding
is accounted for in the max-width. Until they do, quirk the behavior to fix the
issue.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsTextInputBoxSizingBorderBoxQuirk const):
(WebCore::handleLogicMastersQuirks):
(WebCore::Quirks::determineRelevantQuirks):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Canonical link: https://commits.webkit.org/300048@main
